bin/inference/pycbc_inference_plot_inj_recovery

```diff
@@ -0,0 +1,143 @@
+#! /usr/bin/env python
+""" Plots the recovered versus injected parameter values from a population
+of injections.
+"""
+
+import argparse
+import logging
+import matplotlib as mpl
+import matplotlib.colorbar as cbar
+import matplotlib.pyplot as plt
+import numpy
+import pycbc
+from matplotlib import cm
+from pycbc import inject
+from pycbc import transforms
+from pycbc.inference import option_utils
+
+# parse command line
+parser = argparse.ArgumentParser(usage=__file__ + " [--options]",
+                                 description=__doc__)
+parser.add_argument("--output-file", required=True, type=str,
+                    help="Path to save output plot.")
+parser.add_argument("--verbose", action="store_true",
+                    help="Allows print statements.")
+parser.add_argument("--quantiles", nargs="+", type=float, default=[0.05, 0.95])
+parser.add_argument("--injection-hdf-group", default="H1/injections",
+                    help="HDF group that contains injection values.")
+option_utils.add_inference_results_option_group(parser)
+option_utils.add_scatter_option_group(parser)
+opts = parser.parse_args()
+
+# set logging
+pycbc.init_logging(opts.verbose)
+
+# read results
+fp, parameters, labels, samples = option_utils.results_from_cli(opts)
+
+# only plot one parameter
+assert(len(opts.parameters) == 1)
+parameter = parameters[0] if isinstance(parameters, list) else parameters
+label = labels[0] if isinstance(labels, list) else labels
+
+# create figure
+fig = plt.figure()
+ax = fig.add_subplot(111)
+
+# typecast to list for iteratation
+samples = [samples] if not isinstance(samples, list) else samples
+fp = [fp] if not isinstance(fp, list) else fp
+
+# if user wants a colorbar
+if opts.z_arg:
+
+    # store list of z-axis values and label
+    zvals = []
+    zlabel = None
+
+    # loop over input files
+    logging.info("Reading %s values", opts.z_arg)
+    for i, input_fp in enumerate(fp):
+
+        # get z-axis values and label
+        likelihood_stats = input_fp.read_likelihood_stats(
+             thin_start=opts.thin_start, thin_end=opts.thin_end,
+             thin_interval=opts.thin_interval, iteration=opts.iteration)
+        vals, zlabel = option_utils.get_zvalues(input_fp, opts.z_arg,
+                                                likelihood_stats)
+        zvals.append(numpy.median(vals))
+
+        # update range of colorbar
+        min_zval = vals.min() if i == 0 else min(min_zval, vals.min())
+        max_zval = vals.max() if i == 0 else max(max_zval, vals.max())
+
+    # create colormap
+    cmap = cm.get_cmap(opts.scatter_cmap)
+    vmin = opts.vmin if opts.vmin else min_zval
+    vmax = opts.vmax if opts.vmax else max_zval
+    norm = mpl.colors.Normalize(vmin, vmax)
+
+# loop over input files and its samples
+logging.info("Plotting")
+for i, (input_file, input_fp, input_samples) in enumerate(zip(opts.input_file,
```

Comment:
Just a suggestion -- Can input_fp file be renamed to something like inference_hdf_file? That way it might be clearer to users that input_file is the user-provided hdf file and input_hdf_file(or so) is the InferenceFile object.

Reply:
``fp`` is a pretty common abbreviation for file pointer which is basically what you describe.

Reply:
ok